File: java/Tutorials/Common/CS/TutorialUtils.cs
using System;
using System.Diagnostics;
using System.Reflection;

namespace Neurotec.Tutorials
{
	public static class TutorialUtils
	{
		public static void PrintTutorialHeader(string [] args)
		{
			string description = ((AssemblyDescriptionAttribute)Assembly.GetExecutingAssembly().GetCustomAttributes(typeof(AssemblyDescriptionAttribute), false)[0]).Description;
			string version = FileVersionInfo.GetVersionInfo(Assembly.GetExecutingAssembly().Location).FileVersion;
			string copyright = ((AssemblyCopyrightAttribute)Assembly.GetExecutingAssembly().GetCustomAttributes(typeof(AssemblyCopyrightAttribute), false)[0]).Copyright;
			Console.WriteLine(GetAssemblyName());
			Console.WriteLine("");
			Console.WriteLine(@"{0} (Version: {1})", description, version);
			Console.WriteLine(copyright.Replace("?", "(C)"));
			Console.WriteLine();
			if (args != null && args.Length > 0)
			{
				Console.WriteLine("Arguments:");
				foreach (string item in args)
				{
					Console.WriteLine("\t{0}", item);
				}
				Console.WriteLine();
			}
		}

		public static string GetAssemblyName()
		{
			return Assembly.GetExecutingAssembly().GetName().Name;
		}

		public static int PrintException(Exception ex)
		{
			int errorCode = -1;
			Console.WriteLine(ex);

			while ((ex is AggregateException) && (ex.InnerException != null))
				ex = ex.InnerException;

			var neurotecException = ex as INeurotecException;
			if (neurotecException != null)
			{
				errorCode = neurotecException.Code;
			}
			return errorCode;
		}
	}
}

File: java/Tutorials/Common/VB/TutorialUtils.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Diagnostics
Imports System.Reflection

Public NotInheritable Class TutorialUtils
	Private Sub New()
	End Sub
	Public Shared Sub PrintTutorialHeader(ByVal args() As String)
		Dim description As String = (CType(Assembly.GetExecutingAssembly().GetCustomAttributes(GetType(AssemblyDescriptionAttribute), False)(0), AssemblyDescriptionAttribute)).Description
		Dim version As String = FileVersionInfo.GetVersionInfo(Assembly.GetExecutingAssembly().Location).FileVersion
		Dim copyright As String = (CType(Assembly.GetExecutingAssembly().GetCustomAttributes(GetType(AssemblyCopyrightAttribute), False)(0), AssemblyCopyrightAttribute)).Copyright
		Console.WriteLine(GetAssemblyName())
		Console.WriteLine("")
		Console.WriteLine("{0} (Version: {1})", description, version)
		Console.WriteLine(copyright.Replace("?", "(C)"))
		Console.WriteLine()
		If args IsNot Nothing AndAlso args.Length > 0 Then
			Console.WriteLine("Arguments:")
			For Each item As String In args
				Console.WriteLine(vbTab & "{0}", item)
			Next item
			Console.WriteLine()
		End If
	End Sub

	Public Shared Function GetAssemblyName() As String
		Return Assembly.GetExecutingAssembly().GetName().Name
	End Function

	Public Shared Function PrintException(ByVal ex As Exception) As Integer
		Dim errorCode As Integer = -1
		Console.WriteLine(ex)

		Do While (TypeOf ex Is AggregateException) AndAlso (ex.InnerException IsNot Nothing)
			ex = ex.InnerException
		Loop

		Dim neurotecException As INeurotecException = TryCast(ex, INeurotecException)
		If neurotecException IsNot Nothing Then
			errorCode = neurotecException.Code
		End If
		Return errorCode
	End Function
End Class
